kasa/smart/modules/motionsensor.py

```diff
@@ -0,0 +1,37 @@
+"""Implementation of motion sensor module."""
+
+from __future__ import annotations
+
+from ...feature import Feature
+from ..smartmodule import SmartModule
+
+
+class MotionSensor(SmartModule):
+    """Implementation of motion sensor module."""
+
+    REQUIRED_COMPONENT = None  # we depend on availability of key
+    REQUIRED_KEY_ON_PARENT = "detected"
```

Comment:
Is the key present on the parent hub `get_device_info` or is it only on the child T100 device_info?

Reply:
I'd guess it'd only be on the T100 device info. So we should probably either use `_check_supported` or mark `sensitivity` as the dependency for this?

Reply:
How does it work for the contactsensor which also uses that `REQUIRED_KEY_ON_PARENT`?

Re tests we should maybe come up with a way to add the children to empty hub devices and pass them in instead of the child device directly.